Assets/Python/RFCUtils.py

```diff
@@ -378,116 +347,76 @@ def cultureManager(self, tCityPlot, iCulturePercent, iNewOwner, iOldOwner, bBarb
 									
 		#plot
-		for x in range(tCityPlot[0]-1, tCityPlot[0]+2):	# from x-1 to x+1
-			for y in range(tCityPlot[1]-1, tCityPlot[1]+2):	# from y-1 to y+1
-				pPlot = gc.getMap().plot(x, y)
-				
-				iCurrentPlotCulture = pPlot.getCulture(iOldOwner)
+		for (x, y) in self.surroundingPlots(tCityPlot, 1):
+			pPlot = gc.getMap().plot(x, y)
+			
+			iCurrentPlotCulture = pPlot.getCulture(iOldOwner)
 
-				if (pPlot.isCity()):
-					pPlot.setCulture(iNewOwner, iCurrentPlotCulture*iCulturePercent/100, True)
-					pPlot.setCulture(iOldOwner, iCurrentPlotCulture*(100-iCulturePercent)/100, True)
+			if pPlot.isCity():
+				pPlot.setCulture(iNewOwner, iCurrentPlotCulture*iCulturePercent/100, True)
+				pPlot.setCulture(iOldOwner, iCurrentPlotCulture*(100-iCulturePercent)/100, True)
+			else:
+				pPlot.setCulture(iNewOwner, iCurrentPlotCulture*iCulturePercent/3/100, True)
+				pPlot.setCulture(iOldOwner, iCurrentPlotCulture*(100-iCulturePercent/3)/100, True)
+				
+				if bAlwaysOwnPlots:
+					pPlot.setOwner(iNewOwner)
 				else:
-					pPlot.setCulture(iNewOwner, iCurrentPlotCulture*iCulturePercent/3/100, True)
-					pPlot.setCulture(iOldOwner, iCurrentPlotCulture*(100-iCulturePercent/3)/100, True)
-
-				#cut other players culture
-##				for iCiv in range(iNumPlayers):
-##					if (iCiv != iNewOwner and iCiv != iOldOwner):
-##						iPlotCulture = gc.getMap().plot(x, y).getCulture(iCiv)
-##						if (iPlotCulture > 0):
-##							gc.getMap().plot(x, y).setCulture(iCiv, iPlotCulture/3, True)
-							
-				#print (x, y, pPlot.getCulture(iNewOwner), ">", pPlot.getCulture(iOldOwner))
-
-				if (not pPlot.isCity()):
-					if (bAlwaysOwnPlots):
+					if pPlot.getCulture(iNewOwner)*4 > pPlot.getCulture(iOldOwner):
 						pPlot.setOwner(iNewOwner)
-					else:
-						if (pPlot.getCulture(iNewOwner)*4 > pPlot.getCulture(iOldOwner)):
-							pPlot.setOwner(iNewOwner)					
-					#print ("NewOwner", pPlot.getOwner())
+				#print ("NewOwner", pPlot.getOwner())
+						
+			#print (x, y, pPlot.getCulture(iNewOwner), ">", pPlot.getCulture(iOldOwner))
 
 
 
 	#handler
-	def spreadMajorCulture(self, iMajorCiv, iX, iY):  
-		
-		for x in range(iX-3, iX+4):	# from x-4 to x+4
-			for y in range(iY-3, iY+4):	# from y-4 to y+4
-				pPlot = gc.getMap().plot(x, y)
-				if (pPlot.isCity()):
-					city = pPlot.getPlotCity()
-					if (city.getOwner() >= iNumMajorPlayers):
-						iMinor = city.getOwner()
-						iDen = 25
-						if (gc.getPlayer(iMajorCiv).getSettlerValue(x, y) >= 400):
-							iDen = 10
-						elif (gc.getPlayer(iMajorCiv).getSettlerValue(x, y) >= 150):
-							iDen = 15
-							
-						iMinorCityCulture = city.getCulture(iMinor)
-						city.setCulture(iMajorCiv, iMinorCityCulture/iDen, True)
-						
-						iMinorPlotCulture = pPlot.getCulture(iMinor)
-						pPlot.setCulture(iMajorCiv, iMinorPlotCulture/iDen, True)
+	def spreadMajorCulture(self, iMajorCiv, iX, iY):
+		for (x, y) in self.surroundingPlots((iX, iY), 4):

```

Comment:
Looking at the code it is 3 indeed, but the comment said 4. I think I took the value from the comment.

for x in range(iX-3, iX+4):	# from x-4 to x+4